File: library/src/test/java/me/proxer/library/enums/TagTypeTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class TagTypeTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final TagType tagType = api.moshi().adapter(TagType.class).fromJson("\"entry_tag\"");

        assertThat(tagType).isSameAs(TagType.TAG);
    }

    @Test
    void testFallback() throws IOException {
        final TagType tagType = api.moshi().adapter(TagType.class).fromJson("\"xyz\"");

        assertThat(tagType).isSameAs(TagType.OTHER);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/TagTypeTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class TagTypeTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(TagType::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"entry_tag\"") shouldBe TagType.TAG
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe TagType.OTHER
    }
}
